specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2022-02-01/kusto.json

```diff
@@ -4290,7 +4290,7 @@
         },
         "managedIdentityResourceId": {
           "type": "string",
-          "description": "The resource ID of a managed identity (system or user assigned) to be used to authenticate with event hub."
+          "description": "If empty, connection will use Event Hub SAS key.  If Cluser resource ID is provided, the connection will use System Assign Identity.  If User Assigned Identity resource ID is provided, the connection will use that User Assigned Identity."
```

Comment:
1. I wouldn't use "Event Hub SAS key" term, but Non MI based Data connection - you don't want to expose the magic.
2. The structure better be as follows:
Leave empty for Non Managed Identity based data connection. For system assigned identity provide cluster resource Id. For user assigned identity (UAI) provide the UAI resource Id.

Reply:
I've implemented suggested changes.